userspace/libsinsp/sinsp_filtercheck_thread.cpp

```diff
@@ -748,6 +762,21 @@ int32_t sinsp_filter_check_thread::extract_arg(std::string_view fldname,
 		} else {
 			throw sinsp_exception("filter syntax error: " + string(val));
 		}
+	} else if(m_field_id == TYPE_AARGS) {
+		// Separate statement in order to expand future usage
+		// to proc.aargs[1][3]; allow for indexed ancestor args access
```

Comment:
@leogr and @FedeDP TBD if we want to expand this in this PR or another PR, alongside deciding when (this release or the next one). Either way fine from my perspective.

One option could be to simply expand the string arg extraction logic (either using re2 or a custom approach), and assigning the second index that would access the arg of an ancestor to a new variable called m_argid2 or other naming.

WDYT?

Reply:
@leogr couldn't agree more re timing and possible side effect!

I have just rebased this PR!